options.go

```diff
@@ -45,9 +53,54 @@ func WithCredentialsJSON(p []byte) DialerOption {
 	}
 }
 
+// WithDefaultDialOption returns a DialerOption that specifies the default DialOptions used.
+func WithDefaultDialOptions(opts ...DialOption) DialerOption {
+	return func(d *dialerConfig) {
+		d.dialOpts = append(d.dialOpts, opts...)
+	}
+}
+
 // WithTokenSource returns a DialerOption that specifies an OAuth2 token source to be used as the basis for authentication.
 func WithTokenSource(s oauth2.TokenSource) DialerOption {
 	return func(d *dialerConfig) {
 		d.sqladminOpts = append(d.sqladminOpts, apiopt.WithTokenSource(s))
 	}
 }
+
+// A DialOption is an option for configuring how a Dialer's Dial call is executed.
```

Comment:
go/go-style/decisions#doc-comments

Reply:
TIL

Reply:
I was curious about this and went back to Effective Go. This might be a difference with external and internal styleguides. Effective Go doesn't include this detail. https://golang.org/doc/effective_go#commentary